iothub/device/src/Authentication/ClientAuthenticationWithSharedAccessKeyRefreshBehalfOf.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Client.Authentication
+{
+    /// <summary>
+    /// Authentication method that generates shared access signature (SAS) token with refresh, based on a provided shared access key (SAK).
+    /// Build for On Behalf of scenarios only. 
```

Comment:
We should clarify in the comments that this is for gateway scenarios when one client authenticates on behalf of another.